elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala

```diff
@@ -128,8 +128,18 @@ class ElasticSearchQueryProcessor(val logicalPlan: LogicalPlan, val parameters:
   }
 
   private def selectFields(fields: Seq[Attribute], query: SearchDefinition): SearchDefinition = {
-      val stringFields: Seq[String] = fields.map(_.name)
-      query.fields(stringFields.toList: _*)
+      val subDocuments = schemaProvided.toSeq flatMap {
```

Comment:
It seems that multi level subdocuments are not supported. In that case, I'd implement all cases or I'd let it crash in order to let Spark resolve the query.